homeassistant/components/go2rtc/__init__.py

```diff
@@ -188,13 +195,35 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         _LOGGER.warning("Could not connect to go2rtc instance on %s (%s)", url, err)
         return False
 
-    provider = WebRTCProvider(hass, url)
-    async_register_webrtc_provider(hass, provider)
+    entry.runtime_data = Go2RtcData(url, session, client)
+
+    async def inform_cameras() -> None:
+        """Inform camera that go2rtc is available."""
+        # To avoid circular import, we import the camera helper here
```

Comment:
I think we should avoid a circular dependency completely, not just work around the import. we could provide a method for the camera entities to register a listener on the client, and call the listeners here instead.
that way we get a camera -> go2rtc dependency only.

that also removes the need for go2rtc to know about camera-specific logic.

Reply:
it would also remove the need for `get_camera_entities` which is also creating a circular dependency between camera.py and helpers.py